server/middleware/patches.coffee

```diff
@@ -0,0 +1,78 @@
+utils = require '../lib/utils'
+errors = require '../commons/errors'
+wrap = require 'co-express'
+database = require '../commons/database'
+parse = require '../commons/parse'
+{ postPatch } = require './patchable'
+mongoose = require 'mongoose'
+Patch = require '../models/Patch'
+User = require '../models/User'
+
+# TODO: Standardize model names so that this map is unnecessary
+collectionNameMap = {
+  'achievement': 'Achievement'
+  'level_component': 'level.component'
+  'level_system': 'level.system'
+  'thang_type': 'thang.type'
+}
+
+module.exports.post = wrap (req, res) ->
+  # Based on body, figure out what collection and document this patch is for
+  if not req.body.target?.collection
+    throw new errors.UnprocessableEntity('target.collection not provided')
+  collection = req.body.target?.collection
+  modelName = collectionNameMap[collection] or collection
+  try
+    Model = mongoose.model(modelName)
+  catch e
+    if e.name is 'MissingSchemaError'
+      throw new errors.NotFound("#{collection} is not a known model")
+    else
+      throw e
+  if not Model.schema.is_patchable
+    throw new errors.UnprocessableEntity("#{collection} is not patchable")
+    
+  # pass to logic shared with "POST /db/:collection/:handle/patch"
+  yield postPatch(Model, collection)(req, res)
```

Comment:
Cause `postPatch` is also used in `/server/routes/index.coffee`, and the first two args are provided on server setup. But it does seem odd from here... I've changed it to instead to be a curried four-arg function.